src/Confluent.Kafka.Avro/AvroDeserializerExtensions.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright 2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Confluent.SchemaRegistry;
+using Avro.Generic;
+
+
+namespace Confluent.Kafka.AvroSerdes
+{
+    /// <summary>
+    ///     Avro deserialization related extension methods for
+    ///     <see cref="Confluent.Kafka.Consumer" />
+    /// </summary>
+    public static class AvroDeserializerExtensions
+    {
+        /// <summary>
+        ///     Sets the Avro deserializer that will be used to deserialize keys or values with
+        ///     the specified type.
+        /// </summary>
+        /// <param name="consumer">
+        ///     The consumer instance this applies to.
+        /// </param>
+        /// <param name="deserializer">
+        ///     The avro deserializer.
+        /// </param>
+        public static void RegisterAvroDeserializer<T>(this Consumer consumer, AvroDeserializer<T> deserializer)
+        {
+            if (!SerdeState.deserializers.ContainsKey(consumer))
+            {
+                SerdeState.deserializers.Add(consumer, new Dictionary<Type, object>());
+            }
+
+            SerdeState.deserializers[consumer].Add(typeof(T), deserializer);
+        }
+
+
+        /// <summary>
+        ///     Removes the avro deserializer associated with the specified type.
+        /// </summary>
+        /// <param name="consumer">
+        ///     The consumer instance this applies to.
+        /// </param>
+        public static void UnregisterAvroSerializer<T>(this Consumer consumer)
+        {
+            SerdeState.deserializers[consumer].Remove(typeof(T));
```

Comment:
there's detail around cleanup and good hash functions and better error handling that i haven't done, but what's here will do for now.

Reply:
(no longer required)